### PR TITLE
Added algorithm integration docs

### LINKED ISSUE #149: Add algorithm integration documentation
Integrate docs from Jared.

### PR DESCRIPTION
**Pull request by tclarke**
closes #149

### CODE REVIEW

**Comment by wardcr** [comment]
I'm seeing warnings for the :lineno-start option, it looks like that was added in Sphinx 1.3 and we're on 1.2. Maybe we should upgrade or remove those tags?

**Comment by wardcr** [comment]
I made some minor syntax/formatting/typo sorts of fixes. They were systematic and simple enough, I thought it would be easier to just do it myself rather than make PR comments.

It might be good to have someone else review the actual correctness of the instructions, since I'm not familiar with all of the steps, but I didn't notice any issues.

It also would be good to fix the code samples to conform to our coding standards (i.e. PEP-8), but if you want to proceed as is for now due to time, I understand.

File: scale/docs/algorithm_integration/index.rst [hunk 1]
```diff
@@ -0,0 +1,22 @@
+
+.. _algorithm_integration:
+
+================================
+Algorithm Integration into Scale
+================================
+This document explains the steps necessary to integrate an algorithm into the Scale system. The Scale system utilizes
+Docker containers to run algorithms in an isolated environment. This guide will document the steps necessary for getting
+an application ready for Scale integration, creating a docker container and capturing outputs, and putting the Job into
```
Comment (dfaller) [inline scale/docs/algorithm_integration/index.rst]: Capitalize Docker, job and recipe should be lowercase

File: scale/docs/algorithm_integration/step1.rst [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+
+**Algorithms must**:
+
+* **Run standalone without any user inputs**
+    Algorithms must be fully automated.  If your algorithm is prompting for input from a user, the job will continue to
+    wait until it times out.
+* **Fail gracefully**
+    Ideally your algorithm will capture its faults and failures and report an exit code and log an informative message
+    to standard output.  Exit codes for an algorithm can be mapped for debugging and metric purposes.  If failures are
```
Comment (dfaller) [inline scale/docs/algorithm_integration/step1.rst]: Change "output" to "error"

File: scale/docs/algorithm_integration/step1.rst [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+    not captured appropriately, Scale will likely report a timeout or unknown error, which will make debugging your
```
Comment (dfaller) [inline scale/docs/algorithm_integration/step1.rst]: Change "timeout or unknown error" to "general algorithm error"

File: scale/docs/algorithm_integration/step1.rst [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+    algorithm more difficult.
+* **Not display popups**
+    Algorithms must not display error dialogs, file selection menus, splash screens, etc. since there is no user that is
+    able to make a selection or close these windows.  Popup displays will cause an algorithm to hang (since they won't
+    be closed) until the job times out.  This is a common issue with IDL and output will need to be displayed to the
+    standard output instead.
+* **Run on Linux**
+    Any external libraries needed must be compiled for Linux.  Do not bundle your algorithm with Windows DLL's.
```
Comment (dfaller) [inline scale/docs/algorithm_integration/step1.rst]: Remove the apostrophe in DLL's

**Comment by dev-geo** [comment]
FYI, this will likely change as we pursue a Go CLI. Jared is starting that this week. I wouldn't polish the write-up too much. Just make sure the documentation is accurate for now. Thanks!

**Comment by dfaller** [comment]
Trevor, I think we're good to merge.

**Comment by dfaller** [comment]
Trevor, I upgraded the Sphinx version and did some other fixes to get rid of some warnings. I think we're now finally ready to merge this.

**Comment by tclarke** [comment]
I started a go CLI. Tell Jared to look in my fork for the branch.

**Pull request merged**